main.go

```
@@ -5,38 +5,121 @@ import (
 	"fmt"
 	"log"
 	"os"
+	"path/filepath"
 )
 
-var noSpaces bool
+var (
+	cfg      = &Config{}
+	noSpaces bool
+	showHelp bool
+)
+
+func detectTemplate(args []string) (string, error) {
+	switch len(args) {
+	case 0:
+		files, err := os.ReadDir(".")
+		if err != nil {
+			return "", fmt.Errorf("failed to read current directory: %w", err)
+		}
+
+		for _, file := range files {
+			if file.IsDir() {
+				continue
+			}
+			name := file.Name()
+			base := name[:len(name)-len(filepath.Ext(name))]
+			if base == "template" {
+				return name, nil
+			}
+		}
+		return "", nil
```

if no template was detected, why not just return an error?
you're eitherway just treating it as an error at L115

---

No, the check is not needed, because it already returns an error that is caught by the error handling if statement right before